### PR TITLE
ci: New workflow to build wheels

### PR DESCRIPTION
**Pull request by kevjumba**
**What this PR does / why we need it**:
https://github.com/kevjumba/feast/runs/7087711627?check_suite_focus=true => Shows that this workflow works!

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: .github/workflows/build_wheels.yml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+            make install-go-proto-dependencies
+            make install-go-ci-dependencies
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: ./wheelhouse/*.whl
+
+
+  build-python-sdk-macos-py310:
```
Comment (achals) [inline .github/workflows/build_wheels.yml:84]: Why don't we move this into `build-python-sdk`? And have a single step?

File: .github/workflows/build_wheels.yml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        env:
+          RELEASE_VERSION: ${{ steps.get_release_version.outputs.release_version }}
+          VERSION_WITHOUT_PREFIX: ${{ steps.get_release_version_without_prefix.outputs.version_without_prefix }}
+          HIGHEST_SEMVER_TAG: ${{ steps.get_highest_semver.outputs.highest_semver_tag }}
+        run: |
+          echo $RELEASE_VERSION
+          echo $VERSION_WITHOUT_PREFIX
+          echo $HIGHEST_SEMVER_TAG
+
+  build-python-sdk:
```
Comment (achals) [inline .github/workflows/build_wheels.yml:43]: nit, `build-python-wheel`

**Comment by kevjumba** [comment]
still doesn't work fixing

File: .github/workflows/build_wheels.yml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        run: |
+          pip install -U pip setuptools wheel twine
+          make install-protoc-dependencies
+          make install-go-proto-dependencies
+          make install-go-ci-dependencies
+          make build-ui
+      - name: Build py310 wheel
+        if: matrix.os == 'macos-10.15'
+        run: |
+          python3 setup.py sdist bdist_wheel
```
Comment (achals) [inline .github/workflows/build_wheels.yml:69]: You can remove this if you remove `cp310-macosx_x86_64` from `CIBW_SKIP`

File: .github/workflows/build_wheels.yml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      - name: Build UI
+        run: make build-ui
+      - name: Build and install dependencies go dependencies for py310
+        if: matrix.os == 'macos-10.15'
+        run: |
+          pip install -U pip setuptools wheel twine
+          make install-protoc-dependencies
+          make install-go-proto-dependencies
+          make install-go-ci-dependencies
+          make build-ui
```
Comment (achals) [inline .github/workflows/build_wheels.yml:65]: Same comment

File: .github/workflows/build_wheels.yml (hunk 1 above)
Comment (kevjumba) [reply .github/workflows/build_wheels.yml]: done

File: .github/workflows/build_wheels.yml (hunk 2 above)
Comment (kevjumba) [reply .github/workflows/build_wheels.yml]: done.

File: .github/workflows/build_wheels.yml (hunk 3 above)
Comment (kevjumba) [reply .github/workflows/build_wheels.yml]: done

File: .github/workflows/build_wheels.yml (hunk 4 above)
Comment (kevjumba) [reply .github/workflows/build_wheels.yml]: done.

File: .github/workflows/build_wheels.yml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        uses: pypa/cibuildwheel@v2.7.0
+        env:
+          CIBW_BUILD: "cp310-macosx_x86_64"
+          CIBW_ARCHS: "native"
+      - uses: actions/upload-artifact@v2
+        with:
+          name: wheels
+          path: ./wheelhouse/*.whl
+
+  build-source_distribution:
```
Comment (felixwang9817) [inline .github/workflows/build_wheels.yml:88]: nit: use - not _, same for other file

File: .github/workflows/publish.yml [hunk 6]
```diff
@@ -274,13 +208,86 @@ jobs:
           make build-ui
       - name: Build
         run: |
-          python3 setup.py sdist bdist_wheel
-
+          python3 setup.py sdist
```
Comment (felixwang9817) [inline .github/workflows/publish.yml:211]: why are we removing bdist_wheel here?

File: .github/workflows/publish.yml [hunk 7]
```diff
@@ -243,14 +172,19 @@ jobs:
             make install-protoc-dependencies
             make install-go-proto-dependencies
             make install-go-ci-dependencies
-
+      - name: Build py310 specific wheels for macos
```
Comment (felixwang9817) [inline .github/workflows/publish.yml:175]: does this work?

File: .github/workflows/build_wheels.yml (hunk 5 above)
Comment (kevjumba) [reply .github/workflows/build_wheels.yml:88]: done

File: .github/workflows/publish.yml (hunk 7 above)
Comment (kevjumba) [reply .github/workflows/publish.yml:175]: yup

File: .github/workflows/publish.yml (hunk 6 above)
Comment (kevjumba) [reply .github/workflows/publish.yml:211]: we are building it in the py310 build. this is just for the source distribution

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**